contracts/swappers/async/ZRXSwapper.sol

```diff
@@ -39,20 +34,17 @@ contract ZRXSwapper is IZRXSwapper, Swapper {
     uint256, // Max slippage is used off-chain
     bytes calldata _data
   ) internal override returns (uint256 _receivedAmount) {
-    (
-      address inputToken,
-      address outputToken,
-      uint256 inputAmount, // minAmountOut
-      ,
-
-    ) = abi.decode(_data[4:], (address, address, uint256, uint256, Transformation[]));
-
-    require(inputToken == _tokenIn && outputToken == _tokenOut && inputAmount == _amountIn, 'Swapper: incorrect swap information');
-    IERC20(_tokenIn).safeApprove(ZRX, 0);
+    uint256 _initialBalanceTokenIn = IERC20(_tokenIn).balanceOf(address(this));
+    uint256 _initialBalanceOfTokenOut = IERC20(_tokenOut).balanceOf(address(this));
     IERC20(_tokenIn).safeApprove(ZRX, _amountIn);
     (bool success, ) = ZRX.call{value: 0}(_data);
     require(success, 'Swapper: ZRX trade reverted');
-    _receivedAmount = IERC20(_tokenOut).balanceOf(address(this));
+    // Check that token in & amount in was correct
+    require(_initialBalanceTokenIn - IERC20(_tokenIn).balanceOf(address(this)) >= _amountIn, 'Swapper: incorrect swap information');
+    // Check that token out was correct
+    uint256 _finalBalanceOfTokenOut = IERC20(_tokenOut).balanceOf(address(this));
+    _receivedAmount = _finalBalanceOfTokenOut - _initialBalanceOfTokenOut;
+    require(_receivedAmount > 0, 'Swapper: incorrect swap information');
     IERC20(_tokenOut).safeTransfer(_receiver, _receivedAmount);
```

Comment:
```suggestion
    _receivedAmount = IERC20(_tokenOut).balanceOf(address(this)) - _initialBalanceOfTokenOut;
    require(_receivedAmount > 0, 'Swapper: incorrect swap information');
    IERC20(_tokenOut).safeTransfer(_receiver, _receivedAmount);
```

Save the GAS!

Reply:
Believe it or not, solidity compiler optimizes this to the same gas consumption. But will commit de suggestion since `_finalBalanceOfTokenOut` is not used again, so lfg !